### PR TITLE
fault injection filter for envoy config

### PR DESCRIPTION
**Pull request by rshriram**
Reads fault rules from the routing rule spec and sets up fault injection filters in Envoy.

### CODE REVIEW

File: proxy/envoy/config.go [hunk 1]
```diff
@@ -410,7 +421,69 @@ func buildRoutes(svc *model.Service, rulesMap map[string][]*config.RouteRule,
...
+
+	return &AbortFilter{
+		Percent:    int(abortRule.Percent),
+		HTTPStatus: int(abortRule.GetHttpStatus()),
+	}
+}
+
+// buildDelayConfig builds the envoy config related to delay spec in a fault filter
+func buildDelayConfig(delayRule *config.HttpFaultInjection_Delay) *DelayFilter {
+	if delayRule == nil {
```
Comment (ijsnellf) [inline proxy/envoy/config.go]: Could make this more compact:
```golang
if delayRule == nil || delayRule.GetFixedDelay() == nil {
   return nil
}
```

File: proxy/envoy/config.go [hunk 2]
```diff
@@ -410,7 +421,69 @@ func buildRoutes(svc *model.Service, rulesMap map[string][]*config.RouteRule,
...
+	if abortRule == nil {
+		return nil
+	}
+	if abortRule.GetHttpStatus() == 0 {
+		// We do not support gRPC and http2 errors now.
+		return nil
+	}
+
+	return &AbortFilter{
+		Percent:    int(abortRule.Percent),
```
Comment (ijsnellf) [inline proxy/envoy/config.go]: `abortRule.Percent` is a float in the range [0,100] not [0,1]?

**Comment by rshriram** [comment]
So this simple test is not so simple, as we discussed earlier. Its an integration test framework with TPRs in place. And my plan was to do that as the next PR, once I manage to dis-entagle the test framework from driver.go and figure out the format for the TPRs.

So far, we do not provide *any* TPRs to istio. We just leverage existing service definitions in K8S. What we need to do is to define a new test application (not just a->b,t), e.g., a->b(v1),b(v2) or something.. And then submit TPRs and then validate.

File: proxy/envoy/config.go (hunk 2 above)
Comment (rshriram) [inline proxy/envoy/config.go]: Yes, its float in range 0-100.

File: proxy/envoy/config.go (hunk 1 above)
Comment (rshriram) [inline proxy/envoy/config.go]: done

File: proxy/envoy/config.go (hunk 2 above)
Comment (kyessenov) [inline proxy/envoy/config.go]: please test it with a float and an int value. we do several translations and JS is funky about numbers.

**Pull request merged**